internal/schema/schema.go

```diff
@@ -104,6 +105,15 @@ func (s Schema) Hash() (string, error) {
 	return fmt.Sprintf("%x", hashVal), nil
 }
 
+type Extension struct {
+	Name    string
+	Version string
+}
+
+func (e Extension) GetName() string {
+	return e.Name
```

Comment:
To make it easier to support non-public schemas in the future, embed [SchemaQualifiedName](https://github.com/stripe/pg-schema-diff/blob/main/internal/schema/schema.go#L24-L31) in the struct. This will give you the `GetName` method and will make it easier to support cross-schemas extensions.

[Example with Functions
](https://github.com/stripe/pg-schema-diff/blob/main/internal/schema/schema.go#L241-L251)

Reply:
I think if we support non-public schemas in the future, we still wouldn't want to use the SQN. Primarily because we call GetName() in the `Add` method and the schema-aware statement would be 

```
CREATE EXTENSION foobar WITH SCHEMA quux
```

using the extension's name instead of a SQN one. We'd want to add the `Schema string` to the `Extension` struct and only use the SQN when dropping or altering?

Reply:
You'd still want to use schema qualified name, since it stores name and schema. You just wouldn't call get name, and instead you would just plug your fields into the syntax you stated above.

It's probably just a good idea to implement that now! And have the add statement be `CREATE EXTENSION 'bar;' WITH SCHEMA foobar`